src/Ide/Main.hs

```diff
@@ -123,11 +123,15 @@ runLspMode lspArgs@LspArguments{..} idePlugins = do
         hPutStrLn stderr $ "  in directory: " <> dir
         hPutStrLn stderr "If you are seeing this in a terminal, you probably should have run ghcide WITHOUT the --lsp option!"
 
-        runLanguageServer options (pluginHandler plugins) getInitialConfig getConfigFromNotification $ \getLspId event vfs caps wProg wIndefProg _getConfig _rootPath -> do
+        runLanguageServer options (pluginHandler plugins) getInitialConfig getConfigFromNotification $ \getLspId event vfs caps wProg wIndefProg getConfig _rootPath -> do
             t <- t
             hPutStrLn stderr $ "Started LSP server in " ++ showDuration t
             sessionLoader <- loadSession dir
-            -- config <- fromMaybe defaultLspConfig <$> getConfig
+            config <- fromMaybe def <$> getConfig
```

Comment:
What about skip the DefaultConfig constructor, keep the actual definition and only check here that there is no client config doing a pattern matching on `Maybe`.

Reply:
Since we make the parsing always successful, I believe `getConfig` will never return `Nothing`.

Reply:
I'm starting to think that to add alert message that we fallback to default config will require a change in `haskell-lsp` instead around these [lines](https://github.com/alanz/lsp/blob/36b655d9cc538f79b14c847ca383d97ffcc18162/src/Language/LSP/Server/Processing.hs#L358-L363) here. There should be a way for us to specify the default config when parsing error occurs.